src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -883,6 +1020,28 @@ public bool HideAutocompleteMethodOptions
             }
         }
 
+        /// <summary>
+        /// Returns if the current session is Dynamo Revit
+        /// </summary>
+        public bool IsDynamoRevit
+        {
+            get
+            {
+                // HostAnalyticsInfo is not set when this is invoked??
+                //return this.dynamoViewModel.Model.HostAnalyticsInfo.HostName.Equals("Dynamo Revit");
+                var host = this.dynamoViewModel.Model.HostAnalyticsInfo.HostName;
+
+                if (host != null)
+                {
+                    return host.Equals("Dynamo Revit");
```

Comment:
sorry i missed this but i don't think we want to check for the host "Dynamo Revit" by hardcoding it. We have a method GetKnownHosts on PackageManagerClient which can be used to check for the known hosts. @QilongTang What do you think?

Reply:
1. I don't think that package manager method has the same list of names as the host names in analytics, and 
2. It seems a bit fishy to call a service just to get the name Revit/DynamoRevit which is being hardcoded somewhere else.
What will happen if the service is slow, times out etc, how will that impact Dynamo's performance for this simple check?

Reply:
This should be rather simple to achieve. We just need a way to know if we are running under DynamoRevit. It seems it's creating a lot of discussion. Could you, please, point me to a single good solution for it? The goal is to only display a certain user option if Dynamo is used inside Revit. I believe this is the first time something like this is happening (this has been brought up a few times).

To spin this around, if we cannot know if we are under Revit, we will not be able to use this new functionality.